maskrcnn_benchmark/structures/segmentation_mask.py

```diff
@@ -1,51 +1,93 @@
 # Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved.
 import torch
-
+import numpy as np
+from torch.nn.functional import interpolate
 import pycocotools.mask as mask_utils
 
 # transpose
 FLIP_LEFT_RIGHT = 0
 FLIP_TOP_BOTTOM = 1
 
-
 class Mask(object):
     """
     This class is unfinished and not meant for use yet
     It is supposed to contain the mask for an object as
     a 2d tensor
     """
-
-    def __init__(self, masks, size, mode):
-        self.masks = masks
+    def __init__(self, segm, size, mode):
+        width, height = size
+        if isinstance(segm, Mask):
+            mask = segm.mask
+        else:
+            if type(segm) == list:
```

Comment:
I'm not sure we want to support all those different kinds of data in the `Mask` class, specially that we can't convert back and forth between them.
What about letting everything polygon-based (including RLE?) be handled by the Polygon class, and call .convert here if it is a Polygon instance?

Reply:
Current `convert` method in `Polygons` class does not always return a single channel mask. So I'm not sure whether it could be used here. Another option is to write a function outside for converting the format and handle the types.